python/CMakeLists.txt

```diff
@@ -454,7 +454,7 @@ endif()
 add_library(arrow_python SHARED ${PYARROW_CPP_SRCS})
 target_include_directories(arrow_python PUBLIC ${PYARROW_CPP_ROOT_DIR}
                                                ${CMAKE_CURRENT_BINARY_DIR}/pyarrow/src)
-if(NOT CMAKE_VERSION VERSION_LESS 3.16)
+if(ARROW_USE_PRECOMPILED_HEADERS AND NOT CMAKE_VERSION VERSION_LESS 3.16)
```

Comment:
We are supposed to use `cmake_minimum_required(VERSION 3.16)` do we require the `CMAKE_VERSION` check?
On another note, I have taken a look at the rest of the places where we use `target_precompile_headers` and all of them check for `if(ARG_PRECOMPILED_HEADERS AND ARROW_USE_PRECOMPILED_HEADERS)` so this seems like we were missing this check.
Thanks @pitrou

Reply:
Ah, then the CMake check is obsolete indeed. Thanks for noticing!